src/main/java/com/meilisearch/sdk/SearchRequest.java

```diff
@@ -326,174 +330,21 @@ private SearchRequest(
-        return this;
-    }
-
-    /**
-     * Method to set the cropMarker
-     *
-     * @param cropMarker Marker used to crop field values
-     * @return altered SearchRequest
-     */
-    public SearchRequest setCropMarker(String cropMarker) {
-        this.cropMarker = cropMarker;
-        return this;
-    }
-
-    /**
-     * Method to set the highlightPreTag
+     * <p>This method is an alias of setQ()
      *
-     * @param highlightPreTag Highlight tag use before every highlighted query terms
-     * @return altered SearchRequest
-     */
-    public SearchRequest setHighlightPreTag(String highlightPreTag) {
-        this.highlightPreTag = highlightPreTag;
-        return this;
-    }
-
-    /**
-     * Method to set the highlightPostTag
-     *
-     * @param highlightPostTag Highlight tag use after every highlighted query terms
-     * @return altered SearchRequest
-     */
-    public SearchRequest setHighlightPostTag(String highlightPostTag) {
-        this.highlightPostTag = highlightPostTag;
-        return this;
-    }
-
-    /**
-     * Method to set the attributesToHighlight
-     *
-     * @param attributesToHighlight Attributes whose values will contain highlighted matching terms
-     * @return altered SearchRequest
-     */
-    public SearchRequest setAttributesToHighlight(String[] attributesToHighlight) {
-        this.attributesToHighlight = attributesToHighlight;
-        return this;
-    }
-
-    /**
-     * Method to set the filter
-     *
-     * @param filter Filter queries by an attribute value
-     * @return altered SearchRequest
-     */
-    public SearchRequest setFilter(String[] filter) {
-        if (filter != null) {
-            this.filter = filter;
-        }
-        return this;
-    }
-
-    public SearchRequest setFilterArray(String[][] filterArray) {
-        if (filterArray != null) {
-            this.filterArray = filterArray;
-        }
-        return this;
-    }
-
-    /**
-     * Method to set the matches boolean
-     *
-     * @param matches Defines whether an object that contains information about the matches should
-     *     be returned or not
-     * @return altered SearchRequest
-     */
-    public SearchRequest setMatches(boolean matches) {
-        this.matches = matches;
-        return this;
-    }
-
-    /**
-     * Method to set the facetsDistribution
-     *
-     * @param facetsDistribution Facets for which to retrieve the matching count
-     * @return altered SearchRequest
-     */
-    public SearchRequest setFacetsDistribution(String[] facetsDistribution) {
-        this.facetsDistribution = facetsDistribution;
-        return this;
-    }
-
-    /**
-     * Method to set the sort
-     *
-     * @param sort Sort queries by an attribute value
-     * @return altered SearchRequest
+     * @param q Query String
      */
-    public SearchRequest setSort(String[] sort) {
-        this.sort = sort;
-        return this;
+    public void setQuery(String q) {
```

Comment:
I'm not sure if you can return `void` here, since we will have the builder pattern, which is a chain of methods returning the same thing (the updated instance of `this`/`SearchRequest`).